projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -2391,6 +2391,22 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
         }
     }
 
+    /**
+     * Returns the `IgxGridHeaderGroupComponent`'s minimum allloed width.
+     * Used internally for restricting header group component width.
+     * The values below depend on the header cell default right/left padding values.
+	 * @memberof IgxGridBaseComponent
+     */
+    get defaultHeaderGroupMinWidth(): number {
+        if (this.isCosy()) {
+            return 32;
```

Comment:
Could you please set those magic numbers as constants.

Reply:
I will refactor the whole get method once [this PR](https://github.com/IgniteUI/igniteui-angular/pull/3282) is merged.